Build issue

**Describe the bug**
I am reinstalling MOLE on a new work laptop. 


**Laptop (please complete the following information):**
- OS: Ubuntu 24.04.2 LTS
- Compiler: compiler gcc v13.3.0 
- cmake version 3.28.3

[cmake.log](https://github.com/user-attachments/files/20416256/cmake.log)

I followed the instructions on the README and first installed the dependencies with 

`$ sudo apt install cmake libopenblas-dev libeigen3-dev`
this step went well. I went ahead and did `mkdir build && cd build`

But then when I run `cmake` I see the output attached in the `cmake.log` file (output was too long to copy/paste in the GH Issue).

And then `make` fails with:
```
make[2]: *** No rule to make target 'third_party_install/superlu-5.3.0/lib/libsuperlu.a', needed by 'tests/cpp/test1'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:301: tests/cpp/CMakeFiles/test1.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

@JananiPSrinivasan I think the build should fail if the FORTRAN compiler is not found.
The build failed pretty early, and should tell the user so.

From the cmake.log
```
$ cmake ..
-- Detected CXX Compiler ID: GNU
-- Using non-Clang compiler flags.
-- SuperLU CMakeLists.txt after patch: cmake_minimum_required(VERSION 3.10)

Process XSDK defaults ...
USE_XSDK_DEFAULTS = 'FALSE'
-- SuperLU will be built as a static library.
-- The Fortran compiler identification is unknown
CMake Error at CMakeLists.txt:62 (enable_language):
  No CMAKE_Fortran_COMPILER could be found.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,15 @@ else()
     message(STATUS "Using non-Clang compiler flags.")
 endif()
 
+if(UNIX AND NOT APPLE AND NOT MSVC)
+    message(STATUS "Checking for Fortran compiler (gfortran)...")
+    enable_language(Fortran OPTIONAL)
+    if(NOT CMAKE_Fortran_COMPILER)
+        message(FATAL_ERROR "gfortran is required but was not found. Please install it using 'sudo apt install gfortran'.")
+    endif()
+endif()
+
+
 find_package(Eigen3 3.3.7 REQUIRED)
 find_library(OpenBLAS_LIBRARIES NAMES openblas blas PATHS "/usr/lib/x86_64-linux-gnu" "/usr/local/opt/" "usr/local/lib" REQUIRED)
 find_library(LAPACK_LIBRARY lapack REQUIRED PATHS "/usr/lib" "/usr/lib/x86_64-linux-gnu" "/usr/local/lib" "/usr/local/opt/")

